### PR TITLE
refactor util.BuildClusterConfig

### PR DESCRIPTION
**Pull request by ikaven1024**
Signed-off-by: yingjinhui <yingjinhui@didiglobal.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
remove the duplicate code.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm
/assign @RainbowMango

File: pkg/util/membercluster_client.go [hunk 1]
```diff
@@ -111,8 +111,11 @@ func NewClusterDynamicClientSetForAgent(clusterName string, client client.Client
 	return &clusterClientSet, nil
 }
 
-func buildClusterConfig(clusterName string, client client.Client) (*rest.Config, error) {
-	cluster, err := GetCluster(client, clusterName)
+// BuildClusterConfig return rest config for member cluster.
```
Comment (RainbowMango) [inline pkg/util/membercluster_client.go:114]:
```suggestion
// BuildClusterConfig returns the rest config for the member cluster.
```

Comment (RainbowMango) [reply pkg/util/membercluster_client.go:114]: I guess the benefits of using the `getter` thing could make it easier for testing. So, can we add a unit test here?

Comment (ikaven1024) [reply pkg/util/membercluster_client.go:114]: It's covered by `TestNewClusterClientSet` and `TestNewClusterDynamicClientSet`. It seems redundantly~

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**